lib/OptimizationBBO/src/OptimizationBBO.jl

```diff
@@ -44,16 +46,52 @@ function __map_optimizer_args(prob::SciMLBase.OptimizationProblem, opt::BBO;
         mapped_args = (; mapped_args..., MinDeltaFitnessTolerance=abstol)
     end
 
+    if !isnothing(verbose)
+        mapped_args = (; mapped_args..., TraceMode=verbose)
+    end
+
     return mapped_args
 end
 
+function log_lines(rd, progress, maxiters, id)
+    while isopen(rd)
+        ln = readline(rd)
+        isempty(ln) && continue
+        m = match(r"([0-9]*) steps", ln)
+        if !isnothing(m)
+            s = tryparse(Int, m[1])
+            isnothing(s) && continue
+            isnothing(maxiters) && continue
+            progress && @info ln progress=(s / maxiters) _id=id
+        end
+    end
+end
+
+function redirect_output(f::Function, progress, maxiters, id)
```

Comment:
This can be moved to https://github.com/SciML/Optimization.jl/blob/master/src/utils.jl

It might not be needed also (I am not sure, not super familiar with this logging stuff) can't we just use `@withprogress` and stick `log_lines` inside that similar to how this is implemented?

Reply:
And we could then remove the ProgressLogging dependency here (it's already there in Optimization)

Reply:
We can't use `@withprogress` directly because we need to pass the `id` to the `log_lines`. I'll try to come up with a solution that doesn't involve adding another dependency.

Reply:
Just write directly to the logs like is done in OrdinaryDiffEq

Reply:
So the `_id` will be the package name like here?

https://github.com/SciML/OrdinaryDiffEq.jl/blob/924b8312669b798f2762ec376e9ebbd7a7c94e2f/src/solve.jl#L90

Reply:
yes